testing/src/main/java/org/oppia/android/testing/robolectric/OppiaShadowActivityManager.kt

```diff
@@ -0,0 +1,53 @@
+package org.oppia.android.testing.robolectric
+
+import android.app.ActivityManager
+import android.os.Debug
+import org.robolectric.annotation.Implementation
+import org.robolectric.annotation.Implements
+
+/**
+ * Shadows the Activity Manager to extend its testing capabilities.
+ *
+ * There is an existing robolectric shadow of Activity Manager but that doesn't provide us enough
+ * control over Debug.MemoryInfo and hence memory usage can't be tested using that.
+ */
+@Implements(ActivityManager::class)
+class OppiaShadowActivityManager {
```

Comment:
Should we maybe extend Robolectric's ShadowActivityManager to build onto its functionality rather than have our own fork?

Reply:
Yes, good point -- modified this to extend ShadowActivityManager.